tests/cli/base.py

```diff
@@ -74,3 +81,160 @@ def command_test(self, arg, kwarg=None, kwarg2=False):
                 pass
         argv = 'mage.py test'
         self.assertRaises(SystemExit, lambda: manage(dict(test=TestCommand()), argv.split()))
+
+    def test_digest_not_found(self):
+        class TestCommand(Cli):
+            def command_test(self, kwarg=None, kwarg2=False):
+                pass
+        argv = 'mage.py test1:test --kwarg --kwarg2'
+
+        out = StringIO()
+        with patch.object(sys, 'stdout', out):
+            with self.assertRaises(SystemExit):
+                manage(dict(test=TestCommand()), argv.split())
+            self.assertEqual('Commands:\ntest\n', out.getvalue())
+
+    def test_no_command_provided(self):
+        class TestCommand(Cli):
+            def command_test(self, kwarg=None, kwarg2=False):
+                pass
+        argv = 'mage.py'
+
+        out = StringIO()
+        with patch.object(sys, 'stdout', out):
```

Comment:
Why do we catch stdout? Isn't stderr the proper stream to write error message to?